Joe McNally News Challenge Submission

jmcnally17 opened the pull request:
User stories

- [x] User story 1: "I want to see all of today's headlines in one place"
- [x] User story 2: "I want to see a relevant picture to illustrate each news article when I browse headlines"
- [x] User story 3: "I want click a news article title which links to the original article"
- [x] User story 4: "I want to specify a search query on the page and get articles matching this search"
- [ ] User story 5: "I want to see a summarised version of the article "
- [x] User story 6: "I want to read the site comfortably on my phone"
- [x] User story 7: "I want to see whizzy animations in the app"

README checklist

- [x] how to install,
- [x] how to run,
- [x] and how to test

jmcnally17 committed:
Project setup, a simple webpage is shown with a message printed to the console to indicate the webpage is running.

jmcnally17 committed:
Created the model class which starts off with an empty array of articles and can add one as a JS object.

jmcnally17 committed:
Created the view class with the model class dependancy injected into it. The view class uses the model's articles array to display a list of headlines on the webpage.

jmcnally17 committed:
Created the api class which fetches the data from the guardian apis using a personal api key that is hidden from github using environment variables.

jmcnally17 committed:
Created a function to turn an article in the api into a more readable JS object that can be added to the articles array in the model.

jmcnally17 committed:
The view class can now use the api class to fetch the data from the guardian api and display the headlines of the current top 10 news stories.

jmcnally17 committed:
The webpage can display a headlines thumbnail above the headline.

jmcnally17 committed:
The article headline is now a link that sends the user to the article main page on the Guardian's website.

jmcnally17 committed:
Added some css

jmcnally17 committed:
Added a search function where the user can look for articles that contain any of the words they enter. Also a message is displayed when the user's input matches no articles.

jmcnally17 committed:
An error message now displays when the api fails to fetch the articles data.

jmcnally17 committed:
Made the app more mobile friendly by making the width of the body fill the width of the screen for mobile use only.

jmcnally17 committed:
Added an animation for the search field and also allowed the user to simply press enter on their keyboard in order to click the search button.

jmcnally17 committed:
Updated README

jmcnally17 committed:
Updated README

jmcnally17 committed:
Updated README

jmcnally17 committed:
Updated README

jmcnally17 committed:
Updated README

jmcnally17 committed:
Updated README

monenavarro commented on src/articlesView.test.js:
```diff
@@ -0,0 +1,111 @@
+const ArticlesModel = require('./articlesModel');
+const ArticlesApi = require('./articlesApi');
+
+jest.mock('./articlesModel');
```
Great use of mocking your tests.

monenavarro commented on src/articlesView.test.js:
```diff
@@ -0,0 +1,111 @@
+
+    view.displayArticlesFromApi();
+
+    expect(view.api.loadArticles).toHaveBeenCalledTimes(1);
```
V readable code and easy to understand

monenavarro commented on index.html:
```diff
@@ -0,0 +1,15 @@
+  </head>
+  <body>
+    <h1>Guardian Top Stories</h1>
+    <input type="text" id="search-field" placeholder="Search.." />
```
Really cool function/animation that the search bar expands

monenavarro commented on src/articlesModel.js:
```diff
@@ -0,0 +1,32 @@
+class ArticlesModel {
```
Clear naming of classes and functions throughout

monenavarro commented on style.css:
```diff
@@ -0,0 +1,73 @@
+}
+
+h1 {
+  color: white;
```
Colours chosen are comfortable for the eyes and makes the content easy to read on the page

jmcnally17 committed:
Minor Tweaks

Add coverage to .gitignore and add two scripts in package.json for
running jest and jest --coverage.

jmcnally17 committed:
Update CSS and HTML

Create two columns which can both scroll individually. The left column
is for the list of headlines, the right column will be for the article
summary once the user has clicked on the image of an article.

jmcnally17 committed:
Fetch article summary

Change the fields parameter in the fetch URL in ArticlesApi to get the
article summary. Modify the ArticlesModel to add this summary as another
key-value pair in each article object in this.articles.

jmcnally17 committed:
Format with ESLint

jmcnally17 committed:
Add package-lock.json to .gitignore

jmcnally17 committed:
Add article summary option

Add an event listener onto each article thumbnail. When clicked, the
article summary of the one that was clicked should appear in the right
hand column on the news page.

jmcnally17 committed:
Clean up image click

Use a private function to clear the summary column before adding a summary
to it when clicking an image.

Refactor the displayArticles code to extract the adImageClick private
function.

jmcnally17 committed:
CSS for summaries

jmcnally17 committed:
Update README

jmcnally17 committed:
Empty summary column when entering a search

jmcnally17 committed:
Tweak ESLint

jmcnally17 committed:
Remove dotenv dependency

jmcnally17 committed:
Add fade in animations

jmcnally17 committed:
Open links in new tab

jmcnally17 committed:
Change name of ApiKey.js

jmcnally17 committed:
Update README.md

jmcnally17 committed:
Add esbuild as dependency
